serialization/murmur3/README.md

```diff
@@ -12,4 +10,184 @@ Run with:
+<h2>Code Output</h2>
+If we run the example it creates a 3-node cluster and we see output like
+<pre>Key '1', hashcode 768969306, is in partition 31, hosted currently by node node0
+Key '2', hashcode 832990049, is in partition 5, hosted currently by node node0
+</pre>
+Integer key 1 has a hash code of 768969306. The modulus of 768969306 % 271 is 31. So integer key 1 is in partition 31. This is on node 0 of the 3-node cluster.
+
+Key 2 has a hash code 832990049. If we do 832990049 % 271 we get 5. Key 2 is in partition 5. This is also on node 0.
+
+Both keys 1 and 2 are on the same node because that one node is responsible for the two partitions relevant.
+
+Now let’s run the example again:
+<pre>Key '1', hashcode 768969306, is in partition 31, hosted currently by node node1
+Key '2', hashcode 832990049, is in partition 5, hosted currently by node node0
+</pre>
+This creates a fresh 3-node cluster. The output is similar but not identical.
+
+The calculated hash codes for keys 1 and 2 are the same, and therefore so are their partitions. Hashing must be deterministic.
+
+Partition allocation is not deterministic. In the second run, a different node is responsible for partition 31 compared to partition 5, so the keys are on different nodes now.
+
+One last point to look at here is keys 15 and 16.
+<pre>Key '15', hashcode 1398412787, is in partition 213, hosted currently by node node1
+Key '16', hashcode 1099026892, is in partition 213, hosted currently by node node1
+</pre>
+They have different hash codes. When the hash code has modulus 271 applied, keys will end up sharing partitions as we have more keys than partitions. Sometimes this will be consecutive keys.
+
+Since keys 15 and 16 are in the same partition, they will always be on the same node, regardless of the cluster size.
+<h3>Checkpoint</h3>
+As` a bonus, there is one last problem.
+
+In the code,
+<pre>for (int i = 0 ; i &lt; 25 ; i++) {
+    ....
+    byte[] value = new byte[i];
+    iMap.set(i,  value);
+}
+</pre>
+We are iterating through the keys, but the value associated gets bigger. For key 1 the value is 1 byte, for key 2 the value is 2 bytes, and so on. The value for key 25 is 25 bytes, 25 times more than for key 1.
+
+Even if we had 25 Hazelcast servers and it happened that they had 1 of the 25 keys each, there would still be an imbalance as the data records are not the same size.
+
+Normally you would expect variable-sized records to vary within a range rather than continually get larger, but this is a feature of the application. Hazelcast stores what it is given.
+<h2>Summary</h2>
+The use of prime numbers and hashing algorithms make it reasonably likely that a Hazelcast cluster will remain balanced for large data volumes.
+
+It is however not a guarantee.
+
+When having strange patterns to key allocation, it is possible that data allocation might not be uniform. Some partitions may get more records than others.
```

Comment:
Again, you may mention skewed distribution.

Reply:
As above, a cut&paste from https://hazelcast.com/blog/where-are-my-keys/, don't want to amend to keep the two matching